src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -0,0 +1,225 @@
+    (segmentedControl?.querySelector('slot') as HTMLSlotElement)
+      ?.assignedElements()
+      ?.filter((element) =>
+        ['LEO-SEGMENTED-ITEM'].includes(element.tagName)
+      ) ??
+      segmentedControl?.querySelectorAll(
+        ':is(leo-segmented-item)'
+      ) ??
+      []
+  ) as HTMLElement[]
+
+  $: {
+    for (const controlItem of controlItems) {
+      controlItem.setAttribute('tabindex', '0')
+
+      if (controlItem.tagName === 'LEO-SEGMENTED-ITEM') {
+        controlItem.setAttribute('role', 'option')
+
+        if (selected === getValue(controlItem)) {
+          controlItem.setAttribute('aria-selected', '')
+          pillWidth = controlItem.clientWidth;
+          pillPosition = controlItem.offsetLeft;
+        } else controlItem.removeAttribute('aria-selected')
+      } else {
+        controlItem.setAttribute('role', 'menuitem')
+      }
+    }
+  }
+
+  function selectItem(e: Event) {
+    // Find the option which was clicked on, if any.
+    const item = controlItems.find((item) => e.composedPath().includes(item))
+
+    // If the event was triggered for something which isn't an option don't fire
+    // a change event.
+    if (!item) return
+
+    // Use data-is-interactive=true to prevent the menu from closing when selected. This implies
+    // there is interacitivity inside the menu item (e.g. a Toggle), which would be good for the user
+    // to see change state and allowing the user to manually close when ready. In other words, when we want
+    // the styles and menu navigation features of leo-menu-item, but we don't want the auto-close.
+
+    if (item.tagName === 'LEO-SEGMENTED-ITEM') {
+      selected = getValue(item)
+
+      dispatch('select-item', {
+        value: selected
+      })
+    }
+
+    // When using keyboard navigation to call selectItem, ensure click handlers happen on the item,
+    // but not when this was called in a click handler, otherwise we'll get 2x clicks.
+    if (item.tagName === 'LEO-SEGMENTED-ITEM' && e.type !== 'click') {
+      item.click()
+    }
+  }
+
+  let pill: HTMLDivElement;
+
+  onMount(() => {
+    pill.addEventListener("transitionstart", () => {
+      segmentedControl.classList.add("transitioning");
+    });
+
+    pill.addEventListener("transitionend", () => {
+      segmentedControl.classList.remove("transitioning");
+    });
+  });
+</script>
+
+<div
+  bind:this={segmentedControl}
+  class="leo-segmented-control"
+  class:isSmall={size === 'small'}
+  class:isTiny={size === 'tiny'}
+  on:keypress={(e) => {
+    if (e.code !== 'Enter' && e.code !== 'Space') return
+    selectItem(e)
+  }}
+  on:click={selectItem}
+>
+  <div class="pill" style:width={`${pillWidth}px`} style:left={`${pillPosition}px`} bind:this={pill} />
+  <slot />
+</div>
+
+<style lang="scss">
+  .leo-segmented-control {
+    --leo-icon-size: var(--leo-icon-m);
+    --bg: var(--leo-color-container-highlight);
+    --control-padding: var(--leo-spacing-s);
```

Comment:
Probably worth having this configurable - padding is a pretty common tweak.

Reply:
Should it be tweaked, though?

Reply:
The reality with the `Button` has been that it will be (at least a little bit) in different designs. I wouldn't be surprised if it's pretty similar for this